crates/re_query_cache/src/cache.rs

```diff
@@ -0,0 +1,364 @@
+use std::{
+    collections::{BTreeMap, VecDeque},
+    sync::Arc,
+};
+
+use ahash::HashMap;
+use itertools::Either;
+use once_cell::sync::Lazy;
+use parking_lot::RwLock;
+use paste::paste;
+use seq_macro::seq;
+
+use re_data_store::LatestAtQuery;
+use re_log_types::{EntityPath, RowId, StoreId, TimeInt, Timeline};
+use re_query::ArchetypeView;
+use re_types_core::{components::InstanceKey, Archetype, ArchetypeName, Component, ComponentName};
+
+use crate::{ErasedFlatVecDeque, FlatVecDeque};
+
+// ---
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum AnyQuery {
+    LatestAt(LatestAtQuery),
+    // TODO(cmc): range queries support.
+}
+
+impl From<LatestAtQuery> for AnyQuery {
+    fn from(query: LatestAtQuery) -> Self {
+        Self::LatestAt(query)
+    }
+}
+
+// ---
+
+// TODO(cmc): Centralize and harmonize all caches (query, jpeg, mesh).
+static CACHES: Lazy<Caches> = Lazy::new(Caches::default);
+
+/// Maintains the top-level cache mappings.
```

Comment:
Are we expected to have one cache per `DataStore`?

Reply:
`CACHES` cover everything, the `StoreId` is part of the `CacheKey`.

Reply:
Put that in the docstring :)

Reply:
Done in next PR